### PR TITLE
[WIP] JUnit tests for BPMN4TOSCA2BPEL

### PR DESCRIPTION
**Pull request by senle**
Added new Junit Tests.

### CODE REVIEW

**Comment by axfrank** [comment]
Hey @senle,
I think you should remove my commits from the PR, because they are not a part of your work.

**Comment by axfrank** [comment]
Please rebase the commits to one, and remove the  commits from axfrank from the PR

**Comment by axfrank** [comment]
I removed my old files, fixed the merge conflicts (mostly caused by moving from tabs to spaces) and merged with the current master. 

Please do a git rebase.

**Comment by axfrank** [comment]
ruzjozin@gmail.com
latifesen91@gmail.com
Please sign the ECA and fix the merge conflicts.

File: .git-hooks/commit-msg [hunk 1]
```diff
@@ -1,9 +1,9 @@
 #!/bin/sh
 
 test "" = "$(grep '^Signed-off-by: ' "$1" |
-	 sort | uniq -c | sed -e '/^[ 	]*1[ 	]/d')" || {
-	echo >&2 Duplicate Signed-off-by lines.
-	exit 1
+    sort | uniq -c | sed -e '/^[     ]*1[     ]/d')" || {
```
Comment (koppor) [inline .git-hooks/commit-msg]: This seems a tab vs. space issue.

Do not replace them -- Winery decided to keep tabs to reduce issues at the existing PRs.

Do not add files where you changed only tabs to spaces.

File: org.eclipse.winery.bpmn4tosca.converter.tobpel/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#Tue Jun 27 19:00:17 CEST 2017
```
Comment (koppor) [inline org.eclipse.winery.bpmn4tosca.converter.tobpel/gradle/wrapper/gradle-wrapper.properties]: Do not re-add gradle.

We changed from gradle tp Maven also in this submodule.

Maybe, it would be a good idea, just to add your changes to the test class to the commit and throw away the other changes.

1. git gui
2. add files you changed
3. commit
4. `git reset --hard` on the command line

File: org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BPMN4Tosca2BpelTest.java [hunk 3]
```diff
@@ -10,13 +10,15 @@
  * Contributors:
  *     Sebastian Wagner - initial API and implementation
  *     ZTE - support of more gateways
- *******************************************************************************/
+ **************************************************************************/
```
Comment (koppor) [inline org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BPMN4Tosca2BpelTest.java]: Why did you remove some stars?

You should add yourself here. Please carefully check http://eclipse.github.io/winery/dev/ToolChain#github---prepare-pull-request again.

File: org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BPMN4Tosca2BpelTest.java [hunk 4]
```diff
@@ -25,38 +27,239 @@
...
-	public void testTransformGateway()
-			throws ParseException, PlanWriterException, MalformedURLException, URISyntaxException {
-		URI srcUri = Paths.get(RESOURCES_DIR, "bpmn4tosca.exclusivegateway.json").toUri();
-		URI targetUri = Paths.get(RESOURCES_DIR, "managementplan.exclusivegateway.zip").toUri();
-		BPMN4Tosca2BpelTest.class.getResource(".");
-		Bpmn4Tosca2Bpel transformer = new Bpmn4Tosca2Bpel();
-		transformer.transform(srcUri, targetUri);
-	}
+    /**
+     * Timestamp for defining actual Test Results.
```
Comment (koppor) [inline org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BPMN4Tosca2BpelTest.java]: Why is this needed?

Comment (koppor) [inline org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BPMN4Tosca2BpelTest.java]: Please use Java8's new TimeApi: https://jaxenter.de/java-se-8-neuerungen-datetime-api-und-warum-ein-umstieg-auf-java-8-lohnt-13045

LocalDate

File: org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BPMN4Tosca2BpelTest.java [hunk 5]
```diff
@@ -25,38 +27,239 @@
...
+                "yyyy.MM.dd.HH.mm.ss")
+                .format(new Date());
+    }
+
+    /**
+     * Teardown after all Tests.
+     *
+     * @throws Exception
+     *         If something goes wrong.
+     */
```
Comment (koppor) [inline org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BPMN4Tosca2BpelTest.java]: Do not add empty setup and tear down methods

File: org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BPMN4Tosca2BpelTest.java [hunk 6]
```diff
@@ -25,38 +27,239 @@
...
+    @After
+    public void tearDown() throws Exception {
+    }
+
+    /**
+     * Tests the Tranform  Json  to Zip Functionality.
+     *
+     * @throws ParseException
```
Comment (koppor) [inline org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BPMN4Tosca2BpelTest.java]: Convention: No `@throws` documentation at test methods, because this are tests.

File: org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BPMN4Tosca2BpelTest.java [hunk 7]
```diff
@@ -25,38 +27,239 @@
...
+     *
+     * @throws Exception
+     *         If something goes wrong.
+     */
+    @After
+    public void tearDown() throws Exception {
+    }
+
+    /**
+     * Tests the Tranform  Json  to Zip Functionality.
```
Comment (koppor) [inline org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BPMN4Tosca2BpelTest.java]: This should be in the test name --> no need to add a extra comment.

File: org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BPMN4Tosca2BpelTest.java [hunk 8]
```diff
@@ -25,38 +27,239 @@
...
+    public void testTransform() throws ParseException,
+            PlanWriterException, MalformedURLException,
+            URISyntaxException {
+        URI srcUri = Paths.get(RESOURCESDIR,
+                "bpmn4tosca.json").toUri();
+        URI targetUri = Paths.get(RESOURCESDIR,
+                timeStamp + "managementplan.zip").toUri();
+        BPMN4Tosca2BpelTest.class.getResource(".");
+        Bpmn4Tosca2Bpel transformer = new Bpmn4Tosca2Bpel();
+        transformer.transform(srcUri, targetUri);
```
Comment (koppor) [inline org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BPMN4Tosca2BpelTest.java]: What about the test result? Why is the test result not compared with something existing?

File: org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BpelPlanArtefactWriterTest.java [hunk 9]
```diff
@@ -8,47 +8,62 @@
...
+     *         If the Syntax of the URI contains an Error.
+
+     */
+    @Test
+    public void testWritePlan() throws MalformedURLException,
+            ParseException, URISyntaxException {
+        Bpmn4JsonParser parser = new Bpmn4JsonParser();
+        URI uri = Paths.get("src/test/resources/bpmn4tosca/bpmn4tosca.json")
+                .toUri();
+        parser.parse(uri);
```
Comment (koppor) [inline org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BpelPlanArtefactWriterTest.java]: This test does only parse. I think, the method name is contradicting.

File: org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BpelPlanArtefactWriterTest.java [hunk 10]
```diff
@@ -8,47 +8,62 @@
...
-	 * Test method for {@link BpelPlanArtefactWriter#writePlan(ManagementFlow, java.net.URL)}.
-	 */
-	@Test
-	public void testWritePlan() throws MalformedURLException, ParseException, URISyntaxException {
-		Bpmn4JsonParser parser = new Bpmn4JsonParser();
-		URI uri = Paths.get("src/test/resources/bpmn4tosca/bpmn4tosca.json").toUri();
-		//Path testBpmn4JsonFile = Paths.get("C:/temp/bpmn4tosca/bpmn4tosca.json");
-		ManagementFlow mngmtFlow = parser.parse(uri);
-
-//		BpelPlanArtefactWriter writer = new BpelPlanArtefactWriter();
```
Comment (koppor) [inline org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BpelPlanArtefactWriterTest.java]: Why is this not enabled and tested? What did go wrong there?

File: org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/package-info.java [hunk 11]
```diff
@@ -0,0 +1,4 @@
+/**
```
Comment (koppor) [inline org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/package-info.java]: This file is not adding additional information. Please remove it.

File: org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BPMN4Tosca2BpelTest.java [hunk 12]
```diff
@@ -25,38 +27,239 @@
...
+     *         If json contains errors.
+     * @throws PlanWriterException
+     *         if Plan writer has problems.
+     * @throws MalformedURLException
+     *         If file paths are not defined correctly.
+     * @throws URISyntaxException
+     *         If the Syntax of the URI contains an Error.
+     */
+    @Test
+    public void testTransform() throws ParseException,
```
Comment (koppor) [inline org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BPMN4Tosca2BpelTest.java]: Please use self-explantory names. See http://www.vogella.com/tutorials/JUnit/article.html - especially, section 3.4.

**Comment by koppor** [comment]
I commented on each thing, I found. It would be nice to document the tests using markdown.

In the case of multiple authors, plese add `Also-by: Some Bodyelse <somebodyelse@nowhere.com>` for each additional author (instead of multiple `Signed-off-by: ...`). For more information, see <https://www.eclipse.org/projects/handbook/#resources-commit>.

**Comment by senle** [comment]
Current status

The pull request did not pass Eclipse validation.

The following users have invalid Signed-off-by footers. Please use the Signed-Off-By (git commit -s) mechanism to confirm the origin of the submission. The identity in the Signed-off-By must match the Author or the Committer.
alex_frank@email.com

**Comment by koppor** [comment]
The follow-up of this PR is https://github.com/eclipse/winery/pull/88
